extensions/wikia/WikiFactory/WikiFactory.php

```diff
@@ -1214,57 +1214,63 @@ public static function getCurrentStagingHost( $dbName='', $default='', $host = n
 	 * @throws \Exception
 	 */
 	static public function getLocalEnvURL( $url, $forcedEnv = null ) {
-		global $wgWikiaEnvironment, $wgWikiaBaseDomain;
-
 		// first - normalize URL
 		$regexp = '/^http:\/\/([^\/]+)\/?(.*)?$/';
 		if ( preg_match( $regexp, $url, $groups ) === 0 ) {
 			// on fail at least return original url
 			return $url;
 		}
-		$server = $groups[ 1 ];
-		$address = $groups[ 2 ];
+		$server = $groups[1];
+		$path = $groups[2];
 
-		if ( !empty( $address ) ) {
-			$address = '/' . $address;
+		if ( !empty( $path ) ) {
+			$path = '/' . $path;
 		}
 
+		$localServer = self::getLocalEnvServer( $server, $forcedEnv );
+
+		return 'http://' . $localServer . $path;
+	}
+
+	static public function getLocalEnvServer( $server, $forcedEnv = null ) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce the number of returns of this function 6, down to the maximum allowed 3. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1142)